weather_mv/loader_pipeline/bq.py

```diff
@@ -149,10 +156,20 @@ def validate_arguments(cls, known_args: argparse.Namespace, pipeline_args: t.Lis
             logger.info('Region validation completed successfully.')
 
     def __post_init__(self):
-        """Initializes Sink by creating a BigQuery table based on user input."""
+        """Initializes BigQuery table based on user input."""
+        self.project, self.dataset_id, self.table_id = self.output_table.split('.')
+        self.table =  None
+
         if self.zarr:
             self.xarray_open_dataset_kwargs = self.zarr_kwargs
-        with open_dataset(self.first_uri, self.xarray_open_dataset_kwargs,
+
+    def create_bq_table(self, uri: str) -> str:
```

Comment:
Can you explain the reasoning for creating the table in the pipeline and not the `__init__`? I like doing so in the constructor so that we get a signal early on. However, I recognize that this slows down the deployment of the pipeline (when we need to copy and read data).

Reply:
https://github.com/google/weather-tools/blob/ec9a71422f82d34d7e2a7e5497217e87d0b6afa4/weather_mv/loader_pipeline/pipeline.py#L49-L55
because in case of a streaming pipeline, there can be a case that target bucket has no files at all. That will cause `all_uri` to be empty and `first_ur` (which is used to infer the schema would be none too). 
hence putting the table creation stage inside the pipeline as an when a file arrives, it will create the table.

Reply:
Thanks! That is sound reasoning to me. Ok, I'm on board with this approach and will resolve related issues.